neurons/validator/forward.py

```diff
@@ -18,56 +18,73 @@
 from neurons.constants import MOVING_AVERAGE_ALPHA
 from neurons.protocol import ImageGeneration, ImageGenerationTaskModel
 from neurons.utils import colored_log, sh, Stats
-from neurons.validator.backend.client import TensorAlchemyBackendClient
+
 from neurons.validator.backend.exceptions import PostMovingAveragesError
 from neurons.validator.event import EventSchema
-from neurons.validator.rewards.interface import AbstractRewardProcessor
-from neurons.validator.rewards.models.blacklist import BlacklistFilter
-from neurons.validator.rewards.models.nsfw import NSFWRewardModel
 from neurons.validator.rewards.types import MaskedRewards, AutomatedRewards
 from neurons.validator.schemas import Batch
 from neurons.validator.utils import ttl_get_block
+from neurons.validator.config import (
+    get_device,
+    get_metagraph,
+    get_backend_client,
+)
+from neurons.validator.rewards.pipeline import (
+    filter_rewards,
+    get_masked_rewards,
+    get_automated_rewards,
+)
 
 transform = T.Compose([T.PILToTensor()])
 
 
 async def update_moving_averages(
-    validator: "StableValidator",
-    backend_client: TensorAlchemyBackendClient,
+    previous_ma_scores: torch.FloatTensor,
     rewards: torch.Tensor,
-    device: torch.device,
-    alpha=MOVING_AVERAGE_ALPHA,
+    hotkey_blacklist: Optional[List[str]] = None,
+    coldkey_blacklist: Optional[List[str]] = None,
+    alpha: Optional[float] = MOVING_AVERAGE_ALPHA,
 ) -> None:
+    if not hotkey_blacklist:
+        hotkey_blacklist = []
+
+    if not coldkey_blacklist:
+        coldkey_blacklist = []
+
     rewards = torch.nan_to_num(
         rewards,
         nan=0.0,
         posinf=0.0,
         neginf=0.0,
-    ).to(device)
+    ).to(get_device())
+
     moving_average_scores: torch.FloatTensor = alpha * rewards + (
         1 - alpha
-    ) * validator.moving_average_scores.to(device)
+    ) * previous_ma_scores.to(get_device())
 
-    validator.moving_average_scores = moving_average_scores
+    metagraph: bt.metagraph = get_metagraph()
```

Comment:
get_metagraph() params netuid and network default to the test network. The default should probably be production and then we pass the netuid and network here as well in case we want to run on test.